Fable.Remoting.MsgPack/FableConverter.fs

```diff
@@ -0,0 +1,586 @@
+        let valueType = typeof<'v>
+
+        if isDictionary then
+            let dict = Dictionary<'k, 'v> (len)
+
+            for _ in 0 .. len - 1 do
+                dict.Add (read keyType :?> 'k, read valueType :?> 'v)
+
+            box dict
+        else
+            [|
+                for _ in 0 .. len - 1 ->
+                    read keyType :?> 'k, read valueType :?> 'v
+            |] |> Map.ofArray |> box
+
+type ListDeserializer<'a> () =
+    static member Deserialize (len: int, read: Type -> obj) =
+        let argType = typeof<'a>
+
+        [
+            for _ in 0 .. len - 1 ->
+                read argType :?> 'a
+        ]
+#endif
+
+type Reader (data: byte[]) =
+    let mutable pos = 0
+
+    let readInt len m =
+        if BitConverter.IsLittleEndian then
+            let flipped = flip data pos len
+            let x = m (flipped, 0)
+            pos <- pos + len
+            x
+        else
+            pos <- pos + len
+            m (data, pos - len)
+
+    member _.ReadByte () =
+        pos <- pos + 1
+        data.[pos - 1]
+
+    member _.ReadByteArray len =
+        pos <- pos + len
+        data.[ pos - len .. pos - 1 ]
+
+    member _.ReadString len =
+        pos <- pos + len
+        Encoding.UTF8.GetString (data, pos - len, len)
+
+    member x.ReadUInt8 () =
+        x.ReadByte ()
+
+    member x.ReadInt8 () =
+        x.ReadByte () |> sbyte
+
+    member _.ReadUInt16 () =
+        readInt 2 BitConverter.ToUInt16
+
+    member _.ReadInt16 () =
+        readInt 2 BitConverter.ToInt16
+
+    member _.ReadUInt32 () =
+        readInt 4 BitConverter.ToUInt32
+
+    member _.ReadInt32 () =
+        readInt 4 BitConverter.ToInt32
+
+    member _.ReadUInt64 () =
+        readInt 8 BitConverter.ToUInt64
+
+    member _.ReadInt64 () =
+        readInt 8 BitConverter.ToInt64
+
+    member _.ReadFloat32 () =
+        readInt 4 BitConverter.ToSingle
+
+    member _.ReadFloat64 () =
+        readInt 8 BitConverter.ToDouble
+
+    member x.ReadMap (len: int, t: Type) =
+        let args = t.GetGenericArguments ()
+
+        if args.Length <> 2 then
+            failwithf "Expecting %s, but the data contains a map." t.Name
+
+#if !FABLE_COMPILER
+        // todo cache
+        let mapDeserializer = typedefof<DictionaryDeserializer<_,_>>.MakeGenericType args
+        let mapDeserializeMethod = mapDeserializer.GetMethod "Deserialize"
+        let isDictionary = t.GetGenericTypeDefinition () = typedefof<Dictionary<_, _>>
+        
+        mapDeserializeMethod.Invoke (null, [| len; isDictionary; x.Read |])
+#else
+        let pairs =
+            [|
+                for _ in 0 .. len - 1 ->
+                    x.Read args.[0] |> box :?> IComparable, x.Read args.[1]
+            |]
+
+        if t.GetGenericTypeDefinition () = typedefof<Dictionary<_, _>> then
+            let dict = Dictionary<_, _> len
+            pairs |> Array.iter dict.Add
+            box dict
+        else
+            Map.ofArray pairs |> box
```

Comment:
I've hit a bit of a wall here. Map.ofArray fails in IntegrationTestsLive with `t.CompareTo is not a function.` Don't know what else to try.